src/udiskslinuxblock.c

```diff
@@ -575,7 +575,10 @@ find_fstab_entries (UDisksDaemon     *daemon,
         }
       else if (needle != NULL)
         {
-          if (mnt_fs_match_options (fs, needle) == 0)
+          const char *opts;
+
+          opts = mnt_fs_get_options (fs);
+          if (! opts || g_strstr_len (opts, -1, needle) == NULL)
```

Comment:
I think this needs a better explanation? Is something wrong with the `mnt_fs_match_options` function? Shouldn't we report a bug against libmount?

Reply:
Do you need to place this explanation also in the source code?

The thing is that `libmount` somehow classifies mount options into categories and I'm not even sure if all options get considered by the matching functions. E.g. in this case we're looking for custom `x-` prefixed mount option that may be excluded by the parser. It's generally safer to match exact strings like this.

I don't think it's a bug in `libmount`, rather a undefined behaviour. Any fix wouldn't help us due to using already released `util-linux` versions anyway.